### PR TITLE
FIX/Allow frontera to send Requests using methods other than GET

### PR DESCRIPTION
**Pull request by voith**
Fixes vy-labs/crawler#31

### CODE REVIEW

**Comment by voith** [comment]
@RajatGoyal please don't merge this. It's WIP. some manual testing pending. But this is a good start for a discussion.

**Comment by voith** [comment]
Ready for review. Changed title from WIP

File: frontera/contrib/middlewares/fingerprint.py [hunk 1]
```diff
@@ -112,3 +113,19 @@ def _add_fingerprint(self, obj):
...
+    fingerprint_function_name = 'REQUEST_FINGERPRINT_FUNCTION'
+
+    def _get_fingerprint(self, url):
+        return self.fingerprint_function(Request(url))
+
+    def _add_fingerprint(self, obj):
+        request = obj if isinstance(obj, Request) else obj.request
+        obj.meta['fingerprint'] = self.fingerprint_function(request)
+        if 'redirect_urls' in obj.meta:
+            obj.meta['redirect_fingerprints'] = [self._get_fingerprint(url) for url in obj.meta['redirect_urls']]
```
Comment (RajatGoyal) [inline frontera/contrib/middlewares/fingerprint.py:130]: redirect URL fingerprints are already set in the meta, maybe above fingerprint calculations in SQLAlchemyBackend are not required.

Comment (voith) [reply frontera/contrib/middlewares/fingerprint.py:130]: Will verify this. Makes sense though.

Comment (voith) [reply frontera/contrib/middlewares/fingerprint.py:130]: done

**Review by RajatGoyal — APPROVED** [review]

**Pull request merged**